tests/components/openrgb/test_coordinator.py

```diff
@@ -0,0 +1,152 @@
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+async def test_reconnection_on_update_failure(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_openrgb_client: MagicMock,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test that coordinator reconnects when update fails."""
+    mock_config_entry.add_to_hass(hass)
+
+    # Set up the integration
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    # Verify initial state
+    state = hass.states.get("light.ene_dram")
+    assert state
+    assert state.state == STATE_ON
+
+    # Reset mock call counts after initial setup
+    mock_openrgb_client.update.reset_mock()
+    mock_openrgb_client.connect.reset_mock()
+
+    # Simulate the first update call failing, then second succeeding
+    mock_openrgb_client.update.side_effect = [
+        OpenRGBDisconnected(),
+        None,  # Second call succeeds after reconnect
+    ]
+
+    freezer.tick(SCAN_INTERVAL)
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
+
+    # Verify that disconnect and connect were called (reconnection happened)
+    mock_openrgb_client.disconnect.assert_called_once()
+    mock_openrgb_client.connect.assert_called_once()
+
+    # Verify that update was called twice (once failed, once after reconnect)
+    assert mock_openrgb_client.update.call_count == 2
+
+    # Verify that the light is still available after successful reconnect
+    state = hass.states.get("light.ene_dram")
+    assert state
+    assert state.state == STATE_ON
+
+
+async def test_reconnection_fails_second_attempt(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_openrgb_client: MagicMock,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test that coordinator fails when reconnection also fails."""
+    mock_config_entry.add_to_hass(hass)
+
+    # Set up the integration
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    # Verify initial state
+    state = hass.states.get("light.ene_dram")
+    assert state
+    assert state.state == STATE_ON
+
+    # Reset mock call counts after initial setup
+    mock_openrgb_client.update.reset_mock()
+    mock_openrgb_client.connect.reset_mock()
+
+    # Simulate the first update call failing, and reconnection also failing
+    mock_openrgb_client.update.side_effect = [
+        OpenRGBDisconnected(),
+        None,  # Second call would succeed if reconnect worked
+    ]
+
+    # Simulate connect raising an exception to mimic failed reconnection
+    mock_openrgb_client.connect.side_effect = ConnectionRefusedError()
+
+    freezer.tick(SCAN_INTERVAL)
+    async_fire_time_changed(hass)
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
+    await hass.async_block_till_done()
```

Comment:
why do we need to do this 3 times

Reply:
Not quite sure, but we do. Do you want me to investigate that?